core/iwasm/interpreter/wasm_loader.c

```diff
@@ -3153,6 +3174,38 @@ init_llvm_jit_functions_stage2_callback(void *arg)
     os_cond_broadcast(&module->tierup_wait_cond);
     os_mutex_unlock(&module->tierup_wait_lock);
 
+#if WASM_ENABLE_DYNAMIC_PGO != 0
+    {
+        /*
+         * keep monitor function entry counts, wasm_dpgo_get_ent_cnt_value(),
+         * in every second. if a function entry count is larger than the
+         * threshold, tier-up the function.
+         */
+        while (module->orcjit_stop_compiling == false) {
+            sleep(1);
```

Comment:
Had better use a cond and `os_cond_timedwait` it, so that no need to wait 1 second if orcjit_stop_compiling is true.
And seems it is a little long to check every 1 second?

Reply:
1s is just a place holder for functional test. I will tune it with more benchmark.

I think `wasm_loader_unload` will wait for the thread termination. And this tier-up code block seems don't need to quit immediately.
```
    module->orcjit_stop_compiling = true;
    if (module->llvm_jit_init_thread)
        os_thread_join(module->llvm_jit_init_thread, NULL);
```